secboot/secboot_sb.go

```diff
@@ -94,15 +94,23 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 
 	if !res.IsEncrypted {
 		// if we didn't find an encrypted device just return, don't try to
-		// unlock it
-		// the filesystem device for the unencrypted case is the same as the
-		// partition device
+		// unlock it the filesystem device for the unencrypted case is the
+		// same as the partition device
 		res.PartDevice = partDevice
 		res.FsDevice = res.PartDevice
 		return res, nil
 	}
 
-	mapperName := name + "-" + randutilRandomKernelUUID()
+	uuid, err := randutilRandomKernelUUID()
+	if err != nil {
+		// We failed before we could generate the filsystem device path for
+		// the encrypted partition device, so we return FsDevice empty.
+		res.PartDevice = partDevice
```

Comment:
looking at error handling in unlockVolumeUsingSealedKeyTPM this probably should also set IsEncrypted to true on res

Reply:
@pedronis My interpretation of the code at line 95 is that we can only reach this piece of code if `res.IsEncrypted == True`, so it must have been explicitly set before this point. Did I miss something ?